app/assets/javascripts/sdg/sync_goal_and_target_filters.js

```diff
@@ -0,0 +1,28 @@
+(function() {
+  "use strict";
+  App.SDGSyncGoalAndTargetFilters = {
+    sync: function(form) {
+      var goal_filter = form.find("[name*=goal]");
+      var target_filter = form.find("[name*=target]");
+
```

Comment:
I found a case where we are showing all targets when a goal is selected.

1º Go to debates index page
2º Filter by any goal and submit
3º The target selector shows all targets instead of the selected goal ones

We can fix this case by hiding them during initialization:

```
      if (goal_filter.val() > 0) {
        App.SDGSyncGoalAndTargetFilters.enable_goal_options(target_filter, goal_filter.val());
      }
```

Reply:
Nice catch! I've added a fix and tests.